libphal/phal_dump.C

```diff
@@ -99,37 +104,82 @@ struct DumpPPERegValue {
 	}
 } __attribute__((packed));
 
+// Define the chip types numbers
+static constexpr int PROC_SBE_DUMP = 0xA;
+static constexpr int ODYSSEY_SBE_DUMP = 0xB;
+
 /**
- * @brief Return the proc target corresponding to failing unit
- * @param[in] failingUnit Position of the proc containing the failed SBE
+ * @brief Get the Fapi Unit Pos object for OCMB
  *
- * @return Pointer to the pdbg target for proc
+ * @param target The target
+ * @return uint8_t The position
+ */
+uint32_t getFapiUnitPos(pdbg_target* target)
+{
+	uint32_t fapiUnitPos = 0; // chip unit position
+	if (!pdbg_target_get_attribute(target, "ATTR_FAPI_POS", 4, 1,
+				       &fapiUnitPos))
+		log(level::ERROR, "ATTR_FAPI_POS Attribute get failed");
+
+	return fapiUnitPos;
+}
+
+/**
+ * @brief Return the chip target corresponding to failing unit
+ * @param[in] failingUnit Position of the ocmb containing the failed SBE
+ * @param sbeTypeId The chip type number
+ *
+ * @return Pointer to the pdbg target for retreived chip
  *
  * Exceptions: PDBG_TARGET_NOT_OPERATIONAL if target is not available
  */
-struct pdbg_target* getProcFromFailingId(const uint32_t failingUnit)
+struct pdbg_target* getTargetFromFailingId(const uint32_t failingUnit,
+					   const int sbeTypeId)
 {
-	struct pdbg_target* proc = NULL;
-	struct pdbg_target* target;
-	pdbg_for_each_class_target("proc", target)
+	struct pdbg_target* chip = nullptr;
+	struct pdbg_target* target = nullptr;
```

Comment:
no change needed in this line

Reply:
As this line has been moved from LN113 to LN140 thus I thought it would be better to change it in accordance to modern C++ practice

Reply:
Please keep any improvements in separate commits, as the openbmc codereview guidelines, if the change in number of lines are more we will have to make separate commits, please keep a pull request with minimum change for the odyssey to work

Reply:
If the slightest possible improvements (like changing `NULL` to `nullptr`) needed to be made into different commits then the no of commits would be numerous. Also it is very tedious to maintain the integrity of the changing code in that way. It is my opinion and painful experience and I am ready to discuss elaborately in a meeting or in any respective forum.